File: browser/src/main/java/org/sqlunet/browser/xn/DocumentTransformer.java
package org.sqlunet.browser.xn;

import java.io.InputStream;

/**
 * XSL Transformer
 *
 * @author <a href="mailto:1313ou@gmail.com">Bernard Bou</a>
 */
public class DocumentTransformer extends org.sqlunet.browser.web.DocumentTransformer
{
	static private final String XSL_DIR = "/org/sqlunet/";

	/**
	 * Get XSL file
	 *
	 * @param source     type of source
	 * @param isSelector is selector source
	 * @return XSL inputstream
	 */
	protected InputStream getXSLStream(final String source, final boolean isSelector)
	{
		String xsl = null;
		Settings.Source from = Settings.Source.valueOf(source);
		switch (from)
		{
			case WORDNET:
				xsl = DocumentTransformer.XSL_DIR + (isSelector ? "wordnet2html-select.xsl" : "wordnet2html.xsl");
				break;
			case VERBNET:
				xsl = DocumentTransformer.XSL_DIR + (isSelector ? "verbnet2html-select.xsl" : "verbnet2html.xsl");
				break;
			case PROPBANK:
				xsl = DocumentTransformer.XSL_DIR + (isSelector ? "propbank2html-select.xsl" : "propbank2html.xsl");
				break;
			case FRAMENET:
				xsl = DocumentTransformer.XSL_DIR + (isSelector ? "framenet2html-select.xsl" : "framenet2html.xsl");
				break;
			case BNC:
				xsl = DocumentTransformer.XSL_DIR + (isSelector ? "bnc2html-select.xsl" : "bnc2html.xsl");
				break;
		}
		return org.sqlunet.browser.web.DocumentTransformer.class.getResourceAsStream(xsl);
	}
}

File: browser/src/main/java/org/sqlunet/browser/xn/DocumentTransformer.kt
/*
 * Copyright (c) 2023. Bernard Bou
 */
package org.sqlunet.browser.xn

import org.sqlunet.browser.web.DocumentTransformer
import java.io.InputStream

/**
 * XSL Transformer
 *
 * @author [Bernard Bou](mailto:1313ou@gmail.com)
 */
class DocumentTransformer : DocumentTransformer() {

    /**
     * Get XSL file
     *
     * @param from       type of source
     * @param isSelector is selector source
     * @return XSL inputstream
     */
    override fun getXSLStream(from: String, isSelector: Boolean): InputStream? {
        val source = Settings.Source.valueOf(from)
        val xsl: String = when (source) {
            Settings.Source.WORDNET -> XSL_DIR + (if (isSelector) "wordnet2html-select.xsl" else "wordnet2html.xsl")
            Settings.Source.VERBNET -> XSL_DIR + (if (isSelector) "verbnet2html-select.xsl" else "verbnet2html.xsl")
            Settings.Source.PROPBANK -> XSL_DIR + (if (isSelector) "propbank2html-select.xsl" else "propbank2html.xsl")
            Settings.Source.FRAMENET -> XSL_DIR + (if (isSelector) "framenet2html-select.xsl" else "framenet2html.xsl")
            Settings.Source.BNC -> XSL_DIR + (if (isSelector) "bnc2html-select.xsl" else "bnc2html.xsl")
        }
        return DocumentTransformer::class.java.getResourceAsStream(xsl)
    }

    companion object {
        private const val XSL_DIR = "/org/sqlunet/"
    }
}
